Backend/nix/services.nix

```
@@ -0,0 +1,122 @@
+{ inputs, self', ... }:
+{
+  config = {
+    perSystem = { inputs', self', pkgs, lib, ... }: {
+      process-compose."services" = { config, ... }:
+        let
+          startScript = masterDb: replicaDb: port: pkgs.writeShellApplication {
+            name = "start-pgbasebackup";
+            runtimeInputs = with pkgs; [ config.services.postgres."${masterDb}".package coreutils ];
+            text = ''
+              masterDbPath=$(readlink -f data/${masterDb})
+              replicaDbPath=$(readlink -f data/${replicaDb})
+              pg_basebackup -h "$masterDbPath" -U repl_user --checkpoint=fast -D "$replicaDbPath"  -R --slot=some_name  -C --port=${port} 
+            '';
+          };
+        in
+        {
+          imports = [
+            inputs.services-flake.processComposeModules.default
+            inputs.passetto.processComposeModules.default
+          ];
+
+          services.postgres.db-primary = {
+            extensions = extensions: [
+              extensions.postgis
+            ];
+            initialDumps = [
+              ../dev/sql-seed/pre-init.sql
+              ../dev/sql-seed/rider-app-seed.sql
+              ../dev/local-testing-data/rider-app.sql
+              ../dev/sql-seed/public-transport-rider-platform-seed.sql
+              ../dev/local-testing-data/public-transport-rider-platform.sql
+              ../dev/sql-seed/mock-registry-seed.sql
+              ../dev/local-testing-data/mock-registry.sql
+              ../dev/sql-seed/scheduler-example-seed.sql
+              ../dev/sql-seed/dynamic-offer-driver-app-seed.sql
+              ../dev/local-testing-data/dynamic-offer-driver-app.sql
+              ../dev/sql-seed/rider-dashboard-seed.sql
+              ../dev/local-testing-data/rider-dashboard.sql
+              ../dev/sql-seed/provider-dashboard-seed.sql
+              ../dev/local-testing-data/provider-dashboard.sql
+              ../dev/sql-seed/special-zone-seed.sql
+              ../dev/local-testing-data/special-zone.sql
+            ];
+            port = 5432;
+            enable = true;
+            listen_addresses = "*";
+            hbaConf = [
+              { type = "host"; database = "all"; user = "repl_user"; address = "127.0.0.1/32"; method = "trust"; }
+            ];
+            initialScript.before = ''
+              CREATE USER atlas WITH PASSWORD 'atlas';
+              
+              CREATE USER repl_user replication;
+            '';
+          };
+
+          settings.processes.pgBaseBackupForPrimaryDb = {
+            command = "${startScript "db-primary" "db-replica" "5432"}/bin/start-pgbasebackup";
```

```suggestion
            command = startScript "db-primary" "db-replica" 5432;
```

command can be a package; https://github.com/Platonic-Systems/process-compose-flake/blob/9e77499a9f4572d91771b21da17006f24950185c/nix/process-compose/settings/command.nix#L4

---

And likewise in other places below.